tests/test_update.py

```diff
@@ -0,0 +1,168 @@
+# -*- coding: utf-8 -*-
+
+"""
+test_generate_update
+---------------------------------
+"""
+
+from __future__ import unicode_literals
+import logging
+import os
+import pytest
+import git
+import tempfile
+import shutil
+
+from cookiecutter.main import cookiecutter
+from cookiecutter import generate, exceptions, utils
+from cookiecutter.update import DEFAULT_CK_BRANCH
+
+
+@pytest.fixture(scope='function')
+def remove_test_dir(request):
+    """
+    Remove the folder that is created by the test.
+    """
+    def fin_remove_test_dir():
+        if os.path.exists('test_update'):
+            utils.rmtree('test_update')
+    request.addfinalizer(fin_remove_test_dir)
+
+
+@pytest.fixture(scope='function')
+def create_test_dir(request):
```

Comment:
pytest has a `tmpdir` fixture which automatically creates a temporary directory, then later cleans it up. Could it be useful here?

https://docs.pytest.org/en/latest/tmpdir.html

Reply:
I second @barrywhart on using pytest tmpdir fixture. I believe using that fixture will fix Windows build failure.